nasa/ow_autonomy
# Oceanwater 776 flesh out fledgling

#### Overview:
- Added in all Commands and Telemetry from OWLAT Sim to the fledgling.
- Wrote Test sequence plans based off the ones present in the user guide for OWLAT.
- Created new directory for OWLAT plans and changed Cmake so they only compile if the -DOWLAT=ON when built.
- Fixed GUI so that it only displays plans based on what launch file is running.

#### Testing:
- Make sure to clean then build ow_plexil with the -DOWLAT=ON flag.
- In a separate tab start roscore
- In another tab launch OWLAT with ```launch_owlat Scoop```
- In your third tab roslaunch ```roslaunch ow_plexil owlat_exec.launch```
- Run any of the OwlatTestSequences (OwlatTestSequence2 is the quickest) and then OwlatLookupTest to verify the telemetry is working.
- Note that some of the test cases may fail due to OWLAT Sim issues but they shouldn't affect the Plexil plan finishing its execution.

#### Notes 
- There are quite a few different changes in a lot of the files, but the place I would focus most of the reviewing effort is in OwlatAdapter.cpp and OwlatInterface.cpp as that is where the bulk of the changes are.

Thanks!
